sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2118,6 +2129,26 @@ std::pair<RT::PiKernel, std::mutex *> ProgramManager::getOrCreateKernel(
                         &(BuildResult->MBuildResultMutex));
 }
 
+bool doesDevSupportImgAspects(const RTDeviceBinaryImage &Img,
+                              const device &Dev) {
+  const RTDeviceBinaryImage::PropertyRange &PropRange =
+      Img.getDeviceRequirements();
+  for (RTDeviceBinaryImage::PropertyRange::ConstIterator It : PropRange) {
+    using namespace std::literals;
+    if ((*It)->Name != "aspects"sv)
+      continue;
+    ByteArray Aspects = DeviceBinaryProperty(*It).asByteArray();
+    // Drop 8 bytes describing the size of the byte array.
+    Aspects.dropBytes(8);
+    while (!Aspects.empty()) {
+      aspect Aspect = Aspects.consume<aspect>();
+      if (!Dev.has(Aspect))
+        return false;
+    }
+  }
+  return true;
```

Comment:
IIUC `*It` is exactly what range-for/`std::all_of` would provide as the element.

```suggestion
  return std::all_of(PropRange.begin(), PropRange.end(),
                     [&](auto /* Spell out the type if possible */ &Prop) {
    using namespace std::literals;
    if (Prop->Name != "aspects"sv)
      return true;
    ByteArray Aspects = DeviceBinaryProperty(Prop).asByteArray();
    // Drop 8 bytes describing the size of the byte array.
    Aspects.dropBytes(8);
    while (!Aspects.empty()) {
      aspect Aspect = Aspects.consume<aspect>();
      if (!Dev.has(Aspect))
        return false;
    }
    return true;
  });
```